### PR TITLE
fix locking err

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: events/worker.go [hunk 1]
```diff
@@ -10,7 +10,12 @@ type EventLocker func(event *Event) locks.Locker
 
 func nopLocker(_ *Event) locks.Locker { return locks.NopLocker() }
 
-func resourceIDLocker(event *Event) locks.Locker { return locks.KeyLocker(event.ResourceID) }
+func resourceIDLocker(event *Event) locks.Locker {
+	if event.ResourceID == "" {
+		return locks.NopLocker()
+	}
+	return locks.KeyLocker(event.ResourceID)
```
Comment (ibuildthecloud) [inline events/worker.go]: I really think this should be `event.ResourceType + ":" + event.ResourceID`  That is how it's done in the python-agent.

Comment (StrongMonkey) [reply events/worker.go]: but that would be the same string if inspect requests come. What i see in python-agent is that it doesn't use a lock in inspect event. And in go-agent it locks the event but in py-agent it locks the instance.

Comment (StrongMonkey) [reply events/worker.go]: but that would be the same string for inspect events. I have tried that and it still locks the inspect events.

Comment (StrongMonkey) [reply events/worker.go]: I think in py-agent and go agent is that go-agent locks events in the framework which is eventsubcriber, and py-agent locks the events by locking instances in methods like do_instance_activate. Seem like py-agent didn't use locks in do_instance_inspect. Not sure if I understand that correctly.

**Pull request merged**